text/1002-optionality-qualifier.rst

```diff
@@ -0,0 +1,200 @@
+..
+    Status: Draft
+    Type: Guideline
+    Created: 2020-06-17
+    RFC PR: `edgedb/rfcs#0001 <https://github.com/edgedb/rfcs/pull/1>`_
+
+=======================================================
+RFC 1002: Optionality qualifier in properties and links
+=======================================================
+
+This RFC describes the reasons why DDL and SDL declarations for properties
+and links assume optionality by default.
+
+
+Motivation
+==========
+
+The data definition language, and more importantly, the schema definition
+language, which users use to communicate with EdgeDB, need to avoid
+unnecessary verbosity.
+
+This requires some conventions and assumptions about what the user expects
+most of the time. Those assumptions shouldn't bar the user from changing
+their mind when the need arises.
+
+A default value has far reaching consequences which we cover in this
+document.
+
+
+Implementation
+==============
+
+Since the discussion always applies to properties and links, we're referring
+to them simply as *pointers* below.
+
+Assume OPTIONAL qualifiers when no explicit qualifier is provided
+-----------------------------------------------------------------
+
+This is the currently implemented behavior as of EdgeDB v1.0 alpha 3.
+
+An optional pointer is a form of incomplete data that allows the application
+to flexibly decide how to proceed. If the data is there, fine. If it isn't,
+the application can include user-level code to deal with that.
+
+Moreover, multi properties and multi links make the case of an empty set
+intuitively acceptable: since the cardinality allows for any number of
+elements, a state with an empty set is not surprising.
+
+Plenty of examples of existing databases and RPC systems assume optionality
+by default. Relational databases assume optionality unless a NOT NULL
+constraint is passed. Protocol buffers removed support for required fields
+altogether in version 3. JSON-based NoSQL databases like MongoDB assume
+optionality. HTML form fields are optional by default.
```

Comment:
I'd also drop the nod to HTML forms -- pretty much no one uses them as is in modern apps. Don't get me wrong, the argument is still valid, it just feels a bit out of context.